src/container/App.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import './App.css';
+import Display from '../components/Display/Display';
+import ButtonPanel from '../components/ButtonPanel/ButtonPanel';
+import calculate from '../logic/calculate';
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      calculation: '0',
+      total: '',
+      current: '',
+      operator: ''
+    }
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick (btnName) {
+    const data = {
+      calculation: this.state.calculation,
+      total: this.state.total,
+      current: this.state.current,
+      operator: this.state.operator
+    }
+    const newData = calculate(data, btnName);
+    const { calculation, total, current, operator } = newData;
+    this.setState({ calculation, total, current, operator })
```

Comment:
It seems the result of `calculate` is the argument we place in `this.setState`. The argument `data` in `calculate` it seems is the `prevState`. We can thus just declare our whole `handleClick` method as:

```javascript
handleClick(btnName){
  this.setState(prevState => calculate(prevState, btnName))
}
```

Please read on this [article](https://www.freecodecamp.org/news/functional-setstate-is-the-future-of-react-374f30401b6b/) for a short background on functional format of `setState`.